api/src/test/java/org/bahmni/module/events/api/publisher/JMSMessageCreatorTest.java

```diff
@@ -0,0 +1,113 @@
+import static org.bahmni.module.events.api.publisher.JMSMessageCreator.EventHeaderKey.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class JMSMessageCreatorTest {
+
+    @Test
+    public void shouldVerifyUUIDAndIdOfPatientInJsonPayloadGivenEventData() throws JMSException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        Person person = getPerson();
+        Event event = new Event(BAHMNI_PATIENT_CREATED, person, person.getUuid());
+
+        Session session = mock(Session.class);
+        TextMessage message = mock(TextMessage.class);
+
+        JMSMessageCreator jmsMessageCreator = new JMSMessageCreator(objectMapper, event);
+
+        ArgumentCaptor<String> textCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<String> propertyKeyCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<String> propertyValueCaptor = ArgumentCaptor.forClass(String.class);
+
+        when(session.createTextMessage()).thenReturn(message);
+        doNothing().when(message).setText(textCaptor.capture());
+        doNothing().when(message).setStringProperty(propertyKeyCaptor.capture(), propertyValueCaptor.capture());
+
+        jmsMessageCreator.createMessage(session);
+
+        Assertions.assertTrue(textCaptor.getValue().contains("\"uuid\":\"bce786c0-aa57-480d-be6a-23692590086b\""));
+        Assertions.assertTrue(textCaptor.getValue().contains("\"id\":123"));
+    }
+
+    @Test
+    public void shouldVerifyHeadersInPatientEventGivenEventData() throws JMSException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        Person person = getPerson();
+        Event event = new Event(BAHMNI_PATIENT_CREATED, person, person.getUuid());
+
+        Session session = mock(Session.class);
+        TextMessage message = mock(TextMessage.class);
+
+        JMSMessageCreator jmsMessageCreator = new JMSMessageCreator(objectMapper, event);
+
+        ArgumentCaptor<String> propertyKeyCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<String> propertyValueCaptor = ArgumentCaptor.forClass(String.class);
+
+        when(session.createTextMessage()).thenReturn(message);
+        doNothing().when(message).setText(anyString());
+        doNothing().when(message).setStringProperty(propertyKeyCaptor.capture(), propertyValueCaptor.capture());
+
+        jmsMessageCreator.createMessage(session);
+
+        Assertions.assertEquals(EVENT_TYPE.key(), propertyKeyCaptor.getAllValues().get(0));
+        Assertions.assertEquals("BAHMNI_PATIENT_CREATED", propertyValueCaptor.getAllValues().get(0));
+        Assertions.assertEquals(PAYLOAD_ID.key(), propertyKeyCaptor.getAllValues().get(1));
+        Assertions.assertEquals("bce786c0-aa57-480d-be6a-23692590086b", propertyValueCaptor.getAllValues().get(1));
+        Assertions.assertEquals(EVENT_ID.key(), propertyKeyCaptor.getAllValues().get(2));
```

Comment:
Why isn't there a check for non-null / event ID for value of Event ID? Ditto for dateTime?

Reply:
Added the non null check as we can't check the exact value.